drivers: sensor: ams-tmd2635: add initial driver files

yunheQin opened the pull request:
This commit adds initial driver files of ams-tmd2635

This commit modifies the KConfig and CMakeLists.txt in the "drivers/sensor" folder to add compatiable with ams-tmd2635

This commit modifies the i2c.dtsi file in the folder "tests/drivers/sensor"

MaureenHelm reviewed (commented):
This driver looks very similar to the existing tmd2620 driver. Can you enhance the existing driver to support both instead of introducing a new separate driver?

yunheQin commented:
> This driver looks very similar to the existing tmd2620 driver. Can you enhance the existing driver to support both instead of introducing a new separate driver?

Hello @MaureenHelm Firstly, thanks very much for your comments, but I assume that the it is not easy to enhance the existing driver to support both type. I investigate that the datasheet of (tmd2635 & tmd2620), there are many parameter differents, forexample the register address and the data format of the value. Secondly, I also found there are some folder like this topic. So  must I to combine these two driver files?

yunheQin committed:
drivers: sensor: ams-tmd26xx: rename tmd2620 driver files

This commit rename initial driver files of ams-tmd2620
as ams-tmd26xx to support another type sensor

This commit modifies the KConfig and CMakeLists.txt in the
"drivers/sensor" folder to compatiable with ams-tmd26xx

Signed-off-by: Yunhe Qin <yun-he.qin@zeiss.com>

yunheQin committed:
dts: bindings: ams-tmd26xx: add dts bindings for ams proximitory sensor

This commit adds ams,tmd26xx.yaml to dts/bindings/sensor.

Signed-off-by: Yunhe Qin <yun-he.qin@zeiss.com>

yunheQin committed:
drivers: sensor: ams-tmd26xx: modifies initial driver files

This commit modifies initial driver files of ams-tmd26xx
to support ams-tmd2620 and ams-tmd2635

This commit modifies the KConfig and CMakeLists.txt in the
"drivers/sensor" folder to support ams-tmd2620 and
dms-tmd2635

This commit modifies the i2c.dtsi file in the following folder
"tests/drivers/build_all/sensor" to compitable with ams-tmd26xx

This commit modified the file "sensors_trigger_global.conf" and
file "sensors_trigger_none.conf" in the folder
"tests/drivers/build_all/sensor/" to compitable with ams-tmd26xx

This commit rename the file "ams,tmd26xx.yaml" as "ams,tmd26xx-base.yaml"
at the same time the ams,tmd2635.yaml was added.

Signed-off-by: Yunhe Qin <yun-he.qin@zeiss.com>

jilaypandya commented on dts/bindings/sensor/ams,tmd26xx.yaml:
```diff
@@ -0,0 +1,99 @@
+
+compatible: "ams,tmd26xx"
+
+include: [sensor-device.yaml, i2c-device.yaml]
```
you could probably also include ams, tmd2620.yaml over here, and then you need to add only the new properties that are not already present in tmd2620, since tmd2620 properties are a subset of the tmd26xx properties.

jilaypandya commented on drivers/sensor/tmd26xx/tmd26xx.h:
```diff
@@ -0,0 +1,322 @@
+#ifndef ZEPHYR_DRIVERS_SENSOR_TMD26XX_TMD26XX_H_
+#define ZEPHYR_DRIVERS_SENSOR_TMD26XX_TMD26XX_H_
+
+#define TMD2635_CHIP_ID 0X44
```
If possible please move the register defines to tmd26xx_reg.h. Access to these defines might come handy when developing an emul and eventually testing the driver

jilaypandya commented on drivers/sensor/tmd26xx/tmd26xx_trigger.c:
```diff
@@ -1,8 +1,10 @@
 /*
- * Copyright (c) 2023 PHYTEC Messtechnik GmbH
+ * SPDX-FileCopyrightText: Copyright (c) 2023 PHYTEC Messtechnik GmbH
```
dont modify the copyright of PHYTEC Messtechnik GmbH

jilaypandya commented on tests/drivers/build_all/sensor/i2c.dtsi:
```diff
@@ -637,8 +637,8 @@ test_i2c_vl53l1x: vl53l1x@61 {
 
-test_i2c_tmd2620: tmd2620@62 {
-	compatible = "ams,tmd2620";
+test_i2c_tmd26xx: tmd26xx@62 {
```
can you revert this change

jilaypandya commented on drivers/sensor/tmd26xx/Kconfig:
```diff
@@ -0,0 +1,53 @@
+endchoice
+
+config TMD2635
+	bool
```
is this kconfig variable required? you have a choice between the two

yunheQin replied:
this variable is required otherwise the driver will not be initialized

jilaypandya commented on drivers/sensor/tmd26xx/Kconfig:
```diff
@@ -0,0 +1,53 @@
+	  Specify the type of triggering to be used by the driver.
+
+config TMD26XX_SELECT_NONE
+	bool "select none default support tmd2620"
```
SELECT_TMD2620

yunheQin replied:
have modified refer to your proposal
